components/suggest/src/db.rs

```diff
@@ -663,6 +664,57 @@ impl<'a> SuggestDao<'a> {
         Ok(suggestions)
     }
 
+    /// Fetches phantom suggestions
+    pub fn fetch_phantom_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
+        // A single phantom suggestion type can be spread across multiple remote
+        // settings records, for example if it has very many keywords. On ingest
+        // we will insert one row in `phantom_custom_details` and one row in
+        // `suggestions` per record, but that's only an implementation detail.
+        // For consumers, there's only ever at most one phantom suggestion of a
+        // given type.
+        //
+        // Why insert one row in those tables per record? It's how other
+        // suggestions work, and it lets us perform relational operations on
+        // suggestions, records, and keywords. For example, when a record is
+        // deleted we can look up its ID in `suggestions`, join the appropriate
+        // keywords table on the suggestion ID, and delete the keywords that
+        // were added by that record.
+
+        // TODO: Decide on a keywords strategy and look up `query.keyword` in
+        // the appropriate table. For now we return the phantom suggestion whose
+        // type is `query.phantom_suggestion_type` and set `matched_keyword` to
+        // `query.keyword`.
+
+        if let Some(phantom_type) = &query.phantom_suggestion_type {
+            if self.conn.exists(
+                r#"
+                SELECT
+                  s.id
+                FROM
+                  suggestions s
+                JOIN
+                  phantom_custom_details d
+                  ON d.suggestion_id = s.id
+                WHERE
+                  s.provider = :provider
+                  AND d.type = :type
+                "#,
```

Comment:
To double-check by understanding: Once we had the keyword strategy, then there would also be a JOIN and WHERE on the keywords table right?